Out-of-tree builds of integration fail when fortran is enabled

**Describe the bug**
I tried to build the integration subdirectory and I expected it to complete successfully instead errors were encountered around Fortran and nasft.

**GEOPM version**
v3.1

**Expected behavior**
The build completes successfully with a return code of 0.

**Actual behavior**
Steps to reproduce:
1. Build the entire tree: ```GEOPM_SERVICE_CONFIG_OPTIONS="--enable-levelzero --disable-io-uring" ./integration/config/build.sh```
2. Clean the tree: ```cd ${GEOPM_SOURCE}; git clean -ffdx```
3. Try to build integration from the installed artifacts:
```
cd ${GEOPM_SOURCE}/integration
CC=icx CXX=icpx FC=ifx F77=ifx F90=ifx ./configure --with-geopmd=${GEOPM_INSTALL} --with-geopm=${GEOPM_INSTALL} --disable-geopmd-local --disable-geopm-local
make
```
4. Observe the errors:
```
[bgeltz@mcfly integration](release-v3.1)$ make
make  all-am
make[1]: Entering directory '/home/bgeltz/tmp/geopm/integration'
ifx -std=legacy -msse4.2 -I/opt/intel/oneapi/mpi/2021.9.0/include/gfortran/7.1.0 -I/opt/intel/oneapi/mpi/2021.9.0/include  -fiopenmp -fPIC -fPIE -O3 -std=legacy -xAVX -shared-intel -mcmodel=medium -fpic -fiopenmp -fPIC -fPIE  -c -o apps/n
asft/nasft/nas_ft-ft.o `test -f 'apps/nasft/nasft/ft.f90' || echo './'`apps/nasft/nasft/ft.f90
ifx: command line warning #10006: ignoring unknown option '-std=legacy'
ifx: command line warning #10006: ignoring unknown option '-std=legacy'
apps/nasft/nasft/ft.f90(81): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(238): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_EPOCH]
         ierr = geopm_prof_epoch()
----------------^
apps/nasft/nasft/ft.f90(292): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(317): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_REGION]
      ierr = geopm_prof_region(c_char_'AT_PHASE_EVOLVE'//c_null_char, &
-------------^
apps/nasft/nasft/ft.f90(319): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_ENTER]
      ierr = geopm_prof_enter(region_id)
-------------^
apps/nasft/nasft/ft.f90(322): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_INIT]
      ierr = geopm_tprof_init(num_it)
-------------^
apps/nasft/nasft/ft.f90(331): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_POST]
         ierr = geopm_tprof_post()
----------------^
apps/nasft/nasft/ft.f90(335): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_EXIT]
      ierr = geopm_prof_exit(region_id)
-------------^
apps/nasft/nasft/ft.f90(782): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(818): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_REGION]
         ierr = geopm_prof_region(c_char_'AT_PHASE_INDEXMAP_1'//c_null_char, &
----------------^
apps/nasft/nasft/ft.f90(820): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_ENTER]
         ierr = geopm_prof_enter(region_id)
----------------^
apps/nasft/nasft/ft.f90(823): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_INIT]
         ierr = geopm_tprof_init(num_it)
----------------^
apps/nasft/nasft/ft.f90(836): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_POST]
            ierr = geopm_tprof_post()
-------------------^
apps/nasft/nasft/ft.f90(840): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_EXIT]
         ierr = geopm_prof_exit(region_id)
----------------^
apps/nasft/nasft/ft.f90(1074): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(1104): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_REGION]
         ierr = geopm_prof_region(c_char_'AT_PHASE_FFT_1'//c_null_char, &
----------------^
apps/nasft/nasft/ft.f90(1110): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_ENTER]
      ierr = geopm_prof_enter(region_id)
-------------^
apps/nasft/nasft/ft.f90(1113): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_INIT]
      ierr = geopm_tprof_init(num_it)
-------------^
apps/nasft/nasft/ft.f90(1137): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_POST]
         ierr = geopm_tprof_post()
----------------^
apps/nasft/nasft/ft.f90(1141): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_EXIT]
      ierr = geopm_prof_exit(region_id)
-------------^
apps/nasft/nasft/ft.f90(1155): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(1183): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_REGION]
      ierr = geopm_prof_region(c_char_'AT_PHASE_FFT_2'//c_null_char, &
-------------^
apps/nasft/nasft/ft.f90(1185): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_ENTER]
      ierr = geopm_prof_enter(region_id)
-------------^
apps/nasft/nasft/ft.f90(1189): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_INIT]
      ierr = geopm_tprof_init(num_it)
-------------^
apps/nasft/nasft/ft.f90(1213): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_POST]
        ierr = geopm_tprof_post()
---------------^
apps/nasft/nasft/ft.f90(1217): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_EXIT]
      ierr = geopm_prof_exit(region_id)
-------------^
apps/nasft/nasft/ft.f90(1231): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [GEOPM_PROF]
      use geopm_prof
----------^
apps/nasft/nasft/ft.f90(1259): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_REGION]
      ierr = geopm_prof_region(c_char_'AT_PHASE_FFT_3'//c_null_char, &
-------------^
apps/nasft/nasft/ft.f90(1261): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_PROF_ENTER]
      ierr = geopm_prof_enter(region_id)
-------------^
apps/nasft/nasft/ft.f90(1264): error #6404: This name does not have a type, and must have an explicit type.   [GEOPM_TPROF_INIT]
      ierr = geopm_tprof_init(num_it)
-------------^
apps/nasft/nasft/ft.f90(8874): catastrophic error: Too many errors, exiting
compilation aborted for apps/nasft/nasft/ft.f90 (code 1)
make[1]: *** [Makefile:1835: apps/nasft/nasft/nas_ft-ft.o] Error 1
make[1]: Leaving directory '/home/bgeltz/tmp/geopm/integration'
make: *** [Makefile:1318: all] Error 2
```

**Additional Context**
This error can be worked around by disabling fortran at configure time with ``--disable-fortran``.

--- a/integration/configure.ac
+++ b/integration/configure.ac
@@ -248,6 +248,14 @@ if test "x$with_geopm_lib" != x; then
   AC_SUBST([GEOPM_LIBDIR])
 fi
 
+AC_ARG_WITH([geopm-fortran-include], [AS_HELP_STRING([--with-geopm-fortran-include=PATH],
+            [specify directory for the installed geopm Fortran include files.])])
+if test "x$with_geopm_fortran_include" != x; then
+  GEOPM_FC_INCLUDE=$with_geopm_fortran_include
+  AM_FCFLAGS="$AM_FCFLAGS -I$GEOPM_FC_INCLUDE"
+  AC_SUBST([GEOPM_FC_INCLUDE])
+fi
+
 AC_ARG_WITH([libelf], [AS_HELP_STRING([--with-libelf=PATH],
             [specify directory for installed libelf package.])])
 if test "x$with_libelf" != x; then

